### PR TITLE
clarified the use of wildcards in the partition filter

### LINKED ISSUE #25: Correct partition filter documentation
The current partition filter documentation is incorrect. It describes the syntax as defined in HIVE-1609. This was reverted by HIVE-1853 and subsequently revised in HIVE-1862. Specifically, it would be useful to document the supported keywords and associated syntaxes, such as `LIKE` and its `*` wildcard.

It would be great to add a scion somewhere on the [Apache Hive Wiki](https://cwiki.apache.org/confluence/display/Hive/Home), specifically in the section that describes [`SHOW PARTITIONS`](https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL#LanguageManualDDL-ShowPartitions), or wherever this functionality surfaces in the Hive DDL. Once in place, we can link from the Circus Train documentation.

### PR DESCRIPTION
**Pull request by patduin**
fixes #25

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -182,7 +182,7 @@ Partition filters can be specified using the syntax described in [HIVE-1609](htt
 
 > The filter supports "=", "!=", ">", "<", ">=", "<=" and "LIKE" operations on partition keys of type string. "AND" and "OR" logical operations are supported in the filter. So for example, for a table having partition keys country and state, the filter can be 'country = "USA" AND (state = "CA" OR state = "AZ")'
 
-In particular notice that it is possible to nest sub-expressions within parentheses.
+In particular notice that it is possible to nest sub-expressions within parentheses. Wild cards are a allowed in the form of `.*`, example: `year like '201.*'`. Full regular expressions doesn't seem to be supported by Hive.
```
Comment (AnanaMJ) [inline README.md]: "regular expressions doesn't seem..." -> "regular expressions don't seem"

Comment (AnanaMJ) [reply README.md]: Also "are a allowed" -> "are allowed"

Comment (patduin) [reply README.md]: nice catch yes

Comment (abhimanyugupta07) [reply README.md]: She's a grammar wizard ! 💯

Comment (patduin) [reply README.md]: I certainly am not. Thanks for saving my butt!

**Comment by patduin** [comment]
> It would be great to link to this page: https://cwiki.apache.org/confluence/display/Hive/Partition+Filter+Syntax
> 
> This way we don't have to keep our documentation in sync with Hive, and Hive users can generally benefit from having the information in the manual.

That documentation needs some update:
1) Page should move. It's under `Language Manual Select` implying that's it some kind of syntax you can type from the hive cli, it is not, this is the filter that get's passed in the IMetaStoreClient methods (e.g. IMetaStoreClient.listPartitionsByFilter()).
2) It should clearly say this is a partition filter for IMetaStoreClient methods.

**Pull request merged**